### PR TITLE
Adopt plugins without developers

### PR DESCRIPTION
**Pull request by daniel-beck**
Add adopt-this-plugin label to around 330 plugins that do not have maintainers defined in https://github.com/jenkins-infra/repository-permissions-updater

I expect that (almost) all of them are in this state because there hasn't been a release since 2014 (or 2013, don't remember), as that was the cut-off when I introduced per-plugin upload permissions.

Plugins identified using:

    grep -rFH 'developers: []' permissions/ | grep -oE 'plugin-([^.]*)' | cut -c8-

Not all of these plugins may currently be distributed, but 🤷‍♂

This overwhelmingly affects neither popular nor recently uploaded plugins, see below.

### Which of these plugins are installed on more than 5000 (~2% of) instances?

* jquery-ui 37085
* built-on-column 24194
* saferestart 10811
* backup 8602
* discard-old-build 8513
* preSCMbuildstep 5956
* port-allocator 5558
* JDK_Parameter_Plugin 5194
* scp 5374

Determined using:

    for PLUGIN in $( grep -F adopt-this-plugin src/main/resources/label-definitions.properties | sed 's|=.*||g' ) ; do echo "$PLUGIN $( jq '.plugins["'"$PLUGIN"'"]' latestNumbers.json )" ; done

### Which of these plugins have been released since 2013?

Formerly maintained by ndeloof:

* jna-posix-api 2018-06-25
* jnr-posix-api 2018-06-25
* one-shot-executor 2017-01-16
* chromedriver 2015-08-18
* ezwall 2015-03-20
* keep-slave-disconnected 2014-09-25
* oki-docki 2014-11-06

Others

* junit-realtime-test-reporter 2018-04-06 uploaded by @olivergondza (universal upload permissions due to role)
* compound-slaves 2015-12-07 uploaded by `jenkinsadmin` (DEV@cloud)

Determined using:

    for PLUGIN in $( grep -F adopt-this-plugin src/main/resources/label-definitions.properties | sed 's|=.*||g' ) ; do echo "$PLUGIN $( jq '.plugins["'"$PLUGIN"'"].releaseTimestamp' update-center.actual.json )" ; done

### CODE REVIEW

File: src/main/resources/label-definitions.properties [hunk 1]
```diff
@@ -474,35 +516,41 @@ installshield=builder
...
-jcaptcha-plugin=misc
+jboss=misc builder adopt-this-plugin
+jbpm-embedded-plugin=external adopt-this-plugin
+jbpm-workflow-plugin=external adopt-this-plugin
+jcaptcha-plugin=misc adopt-this-plugin
 jclouds-jenkins=slaves cluster cloud
 jdepend=report
-JDK_Parameter_Plugin=parameter
+JDK_Parameter_Plugin=parameter adopt-this-plugin
+jenkins-ci=adopt-this-plugin
```
Comment (oleg-nenashev) [inline src/main/resources/label-definitions.properties]: wut

**Comment by daniel-beck** [comment]
> it is not how the plugin ownership process used to work before

Nothing changes: https://jenkins.io/doc/developer/plugin-governance/adopt-a-plugin/#im-a-plugin-maintainer-and-my-plugin-shows-as-up-for-adoption-why

> P.S: Bonus points for assigning labels via GitHub where possible. I can help with the IRCBot commands for it

You'll have to do that yourself. Note that a minority of affected plugins got labels newly added here, most had one or more labels already, so the overall situation doesn't change much compared to before.

File: src/main/resources/label-definitions.properties (hunk 1 above)
Comment (daniel-beck) [reply src/main/resources/label-definitions.properties]: 🤷‍♂ 

https://github.com/jenkins-infra/repository-permissions-updater/blob/master/permissions/plugin-jenkins-ci.yml

**Comment by daniel-beck** [comment]
I reviewed the list here, checking anything not currently being distributed, and removed two entries that aren't plugins. The rest is:

These plugins are currently suspended:

* AdaptivePlugin
* assembla-oauth
* build-node-column
* buildcoin-plugin
* buildheroes
* caroline
* chrome-frame-plugin
* cifs
* cvs-tag
* dynamicparameter
* emotional-hudson
* external-scheduler
* externalresource-dispatcher
* fortify360
* gcm-notification
* gitorious
* googlecode
* grails
* groovyaxis
* hall-jenkins
* ion-deployer-plugin
* jenkow-activiti-designer
* jenkow-activiti-explorer
* jenkow-plugin
* karotz
* lechat
* liverebel-deploy
* m2-extra-steps
* netio-plugin
* nodeofflinenotification

These are not published for other reasons:

* container-slaves only released as beta in 2015
* heroku-jenkins-plugin only released as beta in 2012

**Comment by oleg-nenashev** [comment]
> Nothing changes: https://jenkins.io/doc/developer/plugin-governance/adopt-a-plugin/#im-a-plugin-maintainer-and-my-plugin-shows-as-up-for-adoption-why

I am not talking about the technical implementation here. We have never marked plugins for adoption without explicit sign-off by maintainers or a best effort to reach out to them. Please correct me if I am wrong.

**Comment by daniel-beck** [comment]
It used to be a wiki label that anyone could set. This is how adopt-a-plugin started, and worked for years, until fairly recently.

Also, by definition, these plugins have had no maintainers for 5+ years. Do you want to wait for an entire decade before acknowledging they're probably unmaintained?

**Comment by oleg-nenashev** [comment]
Yes, my preference to to start a dev list thread and to get it formally approved at the governance meeting on Wednesday. I do not think that 3 days of delay would be a problem there. If you want, I can run this thread to save your time

**Comment by daniel-beck** [comment]
I have no interesting in doing any of that. As I point out above, this is no different from someone batch-editing wiki page labels, which is supported in the current process.

**Comment by daniel-beck** [comment]
I meant to say that I'm not interested in a dev list thread, nothing more. Sorry about the wording.

I missed that the adopt-a-plugin process changes since the wiki version, so I was surprised about your feedback. Now it makes a lot more sense.

I have no interest at this time to pursue this further. Anyone is free to take it over if you think it is worthwhile.

**Comment by oleg-nenashev** [comment]
Thanks for the response @daniel-beck . I will reopen the pull request and assign it to myself in order to eventually handle that. As commented above, I am +1 in principle for marking such unmaintained plugins for adoptions. So I am happy to do the approval legwork there. No ETA, depends on other workstreams and my graphomania

**Comment by daniel-beck** [comment]
@oleg-nenashev Clearing out my open PRs, please base your own PR on this change if you're still interested in this.

**Pull request closed without merging**